### DISCUSSION TITLE
TransferUtility: I can upload/download to the root, but I can't do the same to a folder. Ver 3.7.305.20

### DISCUSSION THREAD

**Discussion by andreig-omnicell** [aws/aws-sdk-net#3168] [Q&A] [answered]
My code 

            var bucketName = "da.test.bucket";
            var profileName = "deafult";
            var accessKey = "***";
            var secretKey = "***";

            regionEndpoint = RegionEndpoint.USWest2;

            Amazon.Util.ProfileManager.RegisterProfile(profileName, accessKey, secretKey);
            var credentials = new StoredProfileAWSCredentials(profileName);            
            var utility = new TransferUtility(new AmazonS3Client(credentials, regionEndpoint));
            var destFileName = "client-test-5.txt";    
            var bfolder = bucketName + "/mydir";
           utility.Upload(sourceFile, bfolder, destFileName);            

Error:
Amazon.S3.AmazonS3Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method. ---> Amazon.Runtime.Internal.HttpErrorResponseException: The remote server returned an error: (403) Forbidden. ---> System.Net.WebException: The remote server returned an error: (403) Forbidden.

When I use 
**var bfolder = bucketName;**
the upload works.

**Comment by dscpinheiro** [comment, marked as answer]
Newer versions of the SDK (after version `3.7.100`) do not allow the bucket name to contain a slash (i.e. part of the object key). See this comment for more context: https://github.com/aws/aws-sdk-net/issues/2622#issuecomment-1581208688

As you mentioned, the solution to this problem is using instead:
```
var destFileName = "mydir/client-test-5.txt"; 
utility.Upload(sourceFile, bucketName, destFileName);
```

This will upload the file to `s3://bucketName/mydir/client-test-5.txt` successfully.
